---
title: "2023_BIochem3BP3_Tutorial"
author: "SLW"
date: "20/11/2023"
output: html_document
---
##Biochem3BP3 Tutorial##

The objective of this tutorial is to learn how to download data from Gene Expression Omnibus (GEO), and perform some visualizations. I also want you to understand how the data was processed and to understand what you are looking at. 

Follow this tutorial and answer questions along with the provided code.

The data downloaded is from this paper. Read paper first to understand the data your are working with: https://academic.oup.com/hmg/article/27/1/135/4581994?login=false

First download the required libraries.
If you do not have these libraries, look up how to download bioconductor packages.

```{r Load needed libraries}
library(GEOquery)
library(GEOmetadb)
library(DBI)
library(limma)
```

```{r Download GEO data}
GEO_data <- as.data.frame(exprs(getGEO("GSE100197")[[1]])) #Takes some time
```

Get to know your data and understand what you are working with. Answer the following questions:

1. What are the dimensions of the data?

2. What type of data structure are you working with? (A) Matrix (B) Data frame (C) Table (D) List

3. What do the columns represent?

4. What do the rows represent?

5. What do the numeric values represent?

```{r Download meta data}
GSE100197 <- getGEO('GSE100197', GSEMatrix = TRUE) #Takes some time
metadata <- pData(phenoData(GSE100197[[1]]))
#select columns
colnames(metadata)
metadata <- pData(phenoData(GSE100197[[1]]))[,c("title","geo_accession","source_name_ch1", "pathology group:ch1", "fetal sex:ch1","gestational age:ch1")]

##Check that meta data and GEO data sample names match
rownames(metadata) == colnames(GEO_data)

#Revalue metadata to the correct data format
metadata$`pathology group:ch1` <- as.factor(metadata$`pathology group:ch1`)
metadata$`fetal sex:ch1` <- as.factor(metadata$`fetal sex:ch1`)
metadata$geo_accession <- as.numeric(metadata$`gestational age:ch1`)

###DO THE REMAINING LABELS ON YOUR OWN
```

##Now work though some exploratory data analysis##
#Try using the dreamRs interface if you are having trouble starting to plot in R#

1. Plot data by the numbers of samples in each pathology group. 
2. Can you identify any clinical variables (e.g. fetal sex, gestational age) that are significantly different between groups? Visualize these data to show differences or lack of differences.
3. Subset the meta data and methylation data to only look at EOPE and Term.
4. How many EOPE samples are there?
5. How many Term samples are there?

With the subsetted data, identify if there are changes in DNA methylation between EOPE and Term
```{r Linear model with DNAm}
#subset the meta data to only EOPE and Term
subset_meta <- metadata[metadata$`pathology group:ch1`=="EOPE" | metadata$`pathology group:ch1` == "Term",]
#remove the factor levels for non-EOPE and Term groups
subset_meta$`pathology group:ch1` <- droplevels(subset_meta$`pathology group:ch1`)

#model design
des <- model.matrix(~subset_meta$`pathology group:ch1`,data = subset_meta)

#linear model
lm <- lmFit(subset, des)
lm_efit <- eBayes(lm)
  
#identify top differentially methylated regions
toptable <- topTable(lm_efit, n = Inf)
toptable_sigificant <- toptable[toptable$adj.P.Val<0.05,]
```

1. How many differenitally methylated CpGs can you identify? Create a plot that shows the differences in DNA methylation data.
2. Explain what your are doing in this linear model and how you interpret the results?
3. Can you identify how you may want to change this model to make it more biologically meaningful? Would you include other variables? Change the samples? Why?
